app/store_loader.go

```diff
@@ -0,0 +1,49 @@
+package app
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/tendermint/tendermint/libs/log"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// StoreLoaderWrapper is a wrapper function that is called before the StoreLoader.
+type StoreLoaderWrapper func(sdk.CommitMultiStore, baseapp.StoreLoader) error
+
+// WrapStoreLoader creates a new StoreLoader by wrapping an existing one.
+func WrapStoreLoader(wrapper StoreLoaderWrapper, storeLoader baseapp.StoreLoader) baseapp.StoreLoader {
+	return func(ms sdk.CommitMultiStore) error {
+		if storeLoader == nil {
+			storeLoader = baseapp.DefaultStoreLoader
+		}
+
+		if wrapper == nil {
+			return errors.New("wrapper must not be nil")
+		}
+
+		return wrapper(ms, storeLoader)
+	}
+}
+
+// PruningWrapper creates a new StoreLoader that first validates the pruning settings before calling the provided StoreLoader.
+func PruningWrapper(logger log.Logger, appOpts servertypes.AppOptions, storeLoader baseapp.StoreLoader) baseapp.StoreLoader {
+	return WrapStoreLoader(func(ms sdk.CommitMultiStore, sl baseapp.StoreLoader) error {
+		const MaxPruningInterval = 13
+
+		// No error checking is needed because we leave that up to the sdk.
+		pruningString, _ := appOpts.Get("pruning-interval").(string)
+		interval, _ := strconv.ParseUint(pruningString, 10, 64)
```

Comment:
Usually we use the `"github.com/spf13/cast"` library when getting things from `appOpts`:

```suggestion
		interval := cast.ToUint64(appOpts.Get("pruning-interval"))
```

`cast.ToUint` should be just fine too if that makes use of `interval` easier. I just have `ToUint64` there because that's what the SDK uses to get the pruning interval from `appOpts` (it's in `server/pruning.go` in the SDK).